x/concentrated-liquidity/internal/math/math.go

```diff
@@ -25,8 +33,14 @@ func Liquidity1(amount sdk.Int, sqrtPriceA, sqrtPriceB sdk.Dec) sdk.Dec {
 	if sqrtPriceA.GT(sqrtPriceB) {
 		sqrtPriceA, sqrtPriceB = sqrtPriceB, sqrtPriceA
 	}
-	diff := sqrtPriceB.Sub(sqrtPriceA)
-	return amount.ToDec().Quo(diff)
+
+	// transform sqrtPriceA and sqrtPriceB to BigDec using osmomath for more final precision
+	sqrtPriceAOsmomath := osmomath.MustNewDecFromStr(sqrtPriceA.String())
+	sqrtPriceBOsmomath := osmomath.MustNewDecFromStr(sqrtPriceB.String())
+	diff := sqrtPriceBOsmomath.Sub(sqrtPriceAOsmomath)
+
+	// conduct the liquidity calculation, then transform back to normal sdk.Dec
+	return osmomath.MustNewDecFromStr(amount.String()).Quo(diff).SDKDec()
```

Comment:
please change conversions

Reply:
Offline, agreed to revert liquidity0 and liquidity1 back to normal sdk.Decs due to minimal improvement in precision